mycelo/loadbot/bot.go

```diff
@@ -40,42 +41,71 @@ func Start(ctx context.Context, cfg *Config) error {
 		return cfg.Accounts[idx].Address, cfg.Amount
 	}
 
-	for _, acc := range cfg.Accounts {
+	// developer accounts / TPS = duration in seconds.
+	// Need the fudger factor to get up a consistent TPS at the target.
+	delay := time.Duration(int(float64(len(cfg.Accounts)*1000/cfg.TransactionsPerSecond)*0.95)) * time.Millisecond
+	startDelay := delay / time.Duration(len(cfg.Accounts))
+
+	for i, acc := range cfg.Accounts {
+		// Spread out client load across different diallers
+		client := cfg.Clients[i%len(cfg.Clients)]
+
+		err := waitFor(ctx, startDelay)
+		if err != nil {
+			return err
+		}
 		acc := acc
 		group.Go(func() error {
-			client, err := cfg.ClientFactory()
-			if err != nil {
-				return err
-			}
-			return runBot(ctx, acc, cfg.TransactionDelay, client, nextTransfer)
+			return runBot(ctx, acc, cfg.Verbose, delay, client, nextTransfer)
 		})
 
 	}
 
 	return group.Wait()
 }
 
-func runBot(ctx context.Context, acc env.Account, sleepTime time.Duration, client bind.ContractBackend, nextTransfer func() (common.Address, *big.Int)) error {
+func runBot(ctx context.Context, acc env.Account, verbose bool, sleepTime time.Duration, client bind.ContractBackend, nextTransfer func() (common.Address, *big.Int)) error {
 	abi := contract.AbiFor("StableToken")
 	stableToken := bind.NewBoundContract(env.MustAddressFor("StableToken"), *abi, client)
 
 	transactor := bind.NewKeyedTransactor(acc.PrivateKey)
 	transactor.Context = ctx
 
+	stableTokenAddress := common.HexToAddress("0xd008")
+	transactor.FeeCurrency = &stableTokenAddress
+
 	for {
-		err := waitFor(ctx, sleepTime)
+		txSentTime := time.Now()
+		recipient, value := nextTransfer()
+		tx, err := stableToken.TxObj(transactor, "transferWithComment", recipient, value, "need to proivde some long comment to make it similar to an encrypted comment").Send()
 		if err != nil {
-			return err
+			if err != context.Canceled {
+				fmt.Printf("Error sending transaction: %v\n", err)
+			}
+			return fmt.Errorf("Error sending transaction: %w", err)
+		}
+		if verbose {
+			fmt.Printf("cusd transfer generated: from: %s to: %s amount: %s\ttxhash: %s\n", acc.Address.Hex(), recipient.Hex(), value.String(), tx.Transaction.Hash().Hex())
+			printJSON(tx)
 		}
 
-		recipient, value := nextTransfer()
-		tx, err := stableToken.TxObj(transactor, "transfer", recipient, value).Send()
+		_, err = tx.WaitMined(ctx)
```

Comment:
if you want for the tx to be mined, then you can't guarantee TPS. a tx will be mined on the next block at most, thus you can only have X transactions per block, where X is the number of dev accounts

why do you want to wait for them to be mined?

Reply:
And also, it really depends on blocktime.. if blocktime is 5s then you can get X txs per 5 seconds

Reply:
I've set up the default load test to have 10k accounts. I wait for the tx b/c if I specify a higher TPS than the block gas limit, I want to slow down the sending of transactions to be similar to the block gas limit.

Reply:
An idea here, probably for the future is to have: `number of tx inflight` per account. So an account can stop sending tx when there's too many tx inflight.

you can do something like that, by watching blocks and reading the txhashes that were included.